### DISCUSSION TITLE
How to use get list entities and row count

### DISCUSSION THREAD

**Discussion by minhhungit** [serenity-is/Serenity#6016] [Help]
Hello team, I need help
I have a custom SqlQuery and it contains 2 sub queries for list entities and count total rows (like List request). But don't know how to execute the sql to 2 variables, how can I do that. I see in ListHandler code we have something like: 

```csharp
query.ForEach(Connection, delegate () { ... });
```

how to use it

**Comment by minhhungit** [comment]
while waiting for the answer, I used this way:
```csharp
using Dapper;

using (var multi = connection.QueryMultiple(query.Text, query.Params))
{
    entities = multi.Read<MyRow>().ToList();
    totalCount = multi.Read<int>().Single();
}
```

**Comment by wezmag** [comment]
I use [this code](https://github.com/serenity-is/common-features/blob/c1b7c3114d403b287b70ee0187f06fe6d196ddf0/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Note/NotesBehavior.cs#L251) as reference and come out with the code below:

``` csharp
public ListResponse<MyRow> ListAll(IDbConnection connection, ListRequest request)
{
    var row = new MyRow();

    var query = new SqlQuery()
                    .From(row)
                    .SelectTableFields()
                    ;

    query.CountRecords = true;
    var entities = new List<MyRow>();
    var totalCount = query.ForEach(connection, () =>
    {
        entities.Add(row);
    });
    return new ListResponse<MyRow>()
    {
        Entities = entities,
        TotalCount = totalCount
    };
}
```
However, I am not quite sure if this is the correct way, but it did get the job done.

**Reply by minhhungit** [reply]
thanks @wezmag
